src/GraphQL.Tests/Serialization/OperationMessageTests.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections.Generic;
+using GraphQL.Transport;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Serialization
+{
+    public class OperationMessageTests : DeserializationTestBase
+    {
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_OperationMessage_Populated(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"id\":\"hello\",\"type\":\"hello2\",\"payload\":{{\"query\":\"hello3\",\"variables\":{ExampleJson}}}}}";
+            var actual = serializer.Deserialize<OperationMessage>(test);
+            actual.Id.ShouldBe("hello");
+            actual.Type.ShouldBe("hello2");
+            actual.Payload.ShouldNotBeNull();
+            var request = serializer.ReadNode<GraphQLRequest>(actual.Payload);
+            request.Query.ShouldBe("hello3");
+            Verify(request.Variables);
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_OperationMessage_Nulls(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"id\":null,\"type\":null,\"payload\":null}}";
+            var actual = serializer.Deserialize<OperationMessage>(test);
+            actual.Id.ShouldBeNull();
+            actual.Type.ShouldBeNull();
+            actual.Payload.ShouldBeNull();
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Reads_OperationMessage_Empty(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{}}";
+            var actual = serializer.Deserialize<OperationMessage>(test);
+            actual.Id.ShouldBeNull();
+            actual.Type.ShouldBeNull();
+            actual.Payload.ShouldBeNull();
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Writes_OperationMessage_Populated(IGraphQLTextSerializer serializer)
+        {
+            var message = new OperationMessage
+            {
+                Id = "hello",
+                Type = "hello2",
+                Payload = new GraphQLRequest
+                {
+                    Query = "hello3",
+                    Variables = new Inputs(new Dictionary<string, object> {
+                        { "arg", ExampleData },
+                    }),
+                }
+            };
+            var actual = serializer.Serialize(message);
+            var expected = $"{{\"type\":\"hello2\",\"id\":\"hello\",\"payload\":{{\"query\":\"hello3\",\"variables\":{{\"arg\":{ExampleJson}}}}}}}";
+            actual.ShouldBeCrossPlatJson(expected);
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void Writes_OperationMessage_Nulls(IGraphQLTextSerializer serializer)
+        {
+            var message = new OperationMessage();
+            var actual = serializer.Serialize(message);
+            var expected = @"{""type"":null}";
```

Comment:
??? Why only type here? Why not `"{}"`?

Reply:
We can change it, but as of now, `OperationMessage` and `GraphQLRequest` always serialize the required fields, even if null.  That is, the 'type' field and the 'query' field respectively.

Reply:
Ah, I remember `if null` code in converters. So why to bother with explicitly serializing _required_ field if its value `null` ?! :)

Reply:
I really don't care.  That's the code I wrote and why I wrote it.  It seems to make sense to me even if just for debugging purposes.  If you'd think something else would be better, we can change it.

Reply:
I think in real life type is never `null` so it does not matter. Let's keep as is.